tests/requests_tests/tests.py

```diff
@@ -1013,10 +1013,27 @@ def test_get_host_suggestion_of_allowed_host(self):
         ):
             request.get_host()
 
-    def test_split_domain_port_removes_trailing_dot(self):
-        domain, port = split_domain_port("example.com.:8080")
-        self.assertEqual(domain, "example.com")
-        self.assertEqual(port, "8080")
+    def test_split_domain_port(self):
+        for host, expected in [
+            ("<invalid>", ("", "")),
+            ("<invalid>:8080", ("", "")),
+            ("example.com 8080", ("", "")),
+            ("example.com:invalid", ("", "")),
+            ("[::1]", ("[::1]", "")),
+            ("[::1]:8080", ("[::1]", "8080")),
```

Comment:
Given that the regex supports IPv6 addresses, shall we add a test for one, such as:

            "[1851:0000:3238:DEF1:0177:0000:0000:0125]:8080"

And I guess we should add a few cases with domain with uppercase?

Reply:
But `[::1]` is an IPv6 address!? Specifically it matches this part of the regex: `\[[a-f0-9]*:[a-f0-9.:]+\]`.

Reply:
I guess we could add in other types of IPv6 address, e.g. an IPv4-mapped IPv6 address, e.g. `[::ffff:127.0.0.1]` which is why there is the `.` in that part of the regex pattern... 🤷🏻

Reply:
Ok. I added both of the above - to ensure that IPv4-mapped IPv6 addresses work as expected. And also your IPv6 address above has uppercase characters in so it tests the existing behavior that lowercases the value.

Reply:
Excellent, thank you! Looks great.